hoomd/hpmc/IntegratorHPMCMono.h

```diff
@@ -1804,6 +2087,907 @@ bool IntegratorHPMCMono<Shape>::py_test_overlap(unsigned int type_i, unsigned in
+            // we sample from their union by checking if any generated position falls in the intersection
+            // of their AABBs, only accepting it if it was generated from neighbor j_min
+
+            // world AABB of particle i
+            detail::AABB aabb_i_old = aabb_i_local_old;
+            aabb_i_old.translate(pos_i_old);
+
+            // extend AABB i by sweep radius
+            range = Scalar(0.5)*d_dep+m_sweep_radius;
+            vec3<Scalar> lower_i = aabb_i_old.getLower();
+            vec3<Scalar> upper_i = aabb_i_old.getUpper();
+            lower_i.x -= range; lower_i.y -= range; lower_i.z -= range;
+            upper_i.x += range; upper_i.y += range; upper_i.z += range;
+
+            // for every pairwise intersection
+            #ifdef ENABLE_TBB
+            tbb::parallel_for(tbb::blocked_range<unsigned int>(0, (unsigned int)intersect_i.size()),
+                [=, &intersect_i, &image_i, &aabbs_i,
+                    &accept, &rng_depletants_parallel,
+                    &thread_counters, &thread_implicit_counters](const tbb::blocked_range<unsigned int>& s) {
+            for (unsigned int k = s.begin(); k != s.end(); ++k)
+            #else
+            for (unsigned int k = 0; k < intersect_i.size(); ++k)
+            #endif
+                {
+                // world AABB of shape j, in image of i
+                detail::AABB aabb_j = aabbs_i[k];
+
+                // extend AABB j by sweep radius
+                vec3<Scalar> lower_j = aabb_j.getLower();
+                vec3<Scalar> upper_j = aabb_j.getUpper();
+                lower_j.x -= range; lower_j.y -= range; lower_j.z -= range;
+                upper_j.x += range; upper_j.y += range; upper_j.z += range;
+
+                // we already know the AABBs are overlapping, compute their intersection
+                vec3<Scalar> intersect_lower, intersect_upper;
+                intersect_lower.x = std::max(lower_i.x, lower_j.x);
+                intersect_lower.y = std::max(lower_i.y, lower_j.y);
+                intersect_lower.z = std::max(lower_i.z, lower_j.z);
+                intersect_upper.x = std::min(upper_i.x, upper_j.x);
+                intersect_upper.y = std::min(upper_i.y, upper_j.y);
+                intersect_upper.z = std::min(upper_i.z, upper_j.z);
+
+                detail::AABB aabb_intersect(intersect_lower, intersect_upper);
+
+                // intersectionAABB volume
+                Scalar V =  (intersect_upper.x-intersect_lower.x)*(intersect_upper.y-intersect_lower.y)*(intersect_upper.z-intersect_lower.z);
+
+                // chooose the number of depletants in the intersection volume
+                hoomd::PoissonDistribution<Scalar> poisson(m_fugacity[type]*V);
+                #ifdef ENABLE_TBB
+                hoomd::RandomGenerator& my_rng = rng_depletants_parallel.local();
+                #else
+                hoomd::RandomGenerator& my_rng = rng_depletants;
+                #endif
```

Comment:
Can we simplify this to
```
#ifndef ENABLE_TBB
hoomd::RandomGenerator& my_rng = rng_depletants;
#endif
```

and then remove the `#else` component of the `ifdef` in the loop below? Basically, initialize the rng outside the loop if we're not using TBB, otherwise initialize it inside?

Reply:
I don't see how that would simplify the code ...